app/src/main/java/com/imashnake/animite/features/searchbar/SearchViewModel.kt

```diff
@@ -1,44 +1,71 @@
 package com.imashnake.animite.features.searchbar
 
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.imashnake.animite.data.Resource
 import com.imashnake.animite.data.repos.SearchRepository
-import com.imashnake.animite.features.searchbar.SearchUiState
+import com.imashnake.animite.dev.ext.string
+import com.imashnake.animite.type.MediaFormat
 import com.imashnake.animite.type.MediaType
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.Job
-import kotlinx.coroutines.launch
-import java.io.IOException
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
 import javax.inject.Inject
 
+@OptIn(ExperimentalCoroutinesApi::class)
 @HiltViewModel
 class SearchViewModel @Inject constructor(
-    private val searchRepository: SearchRepository
+    private val searchRepository: SearchRepository,
+    private val savedStateHandle: SavedStateHandle
 ): ViewModel() {
-    var uiState by mutableStateOf(SearchUiState())
-        private set
+    private val mediaType = savedStateHandle.getStateFlow<MediaType?>("mediaType", null)
```

Comment:
Hardcoded strings make me ill

Reply:
What do I do instead?

Reply:
```
companion object {
  private const val MEDIA_TYPE = "mediaType"
}

private val mediaType = savedStateHandle.getStateFlow<MediaType?>(MEDIA_TYPE, null)
```
in the VM, or in the Constants file if it has any usages anywhere else